structuretoolkit/analyse/snap.py

```diff
@@ -0,0 +1,397 @@
+        ]
+    }
+    d.update(
+        {
+            (k + str(i + 1)): bispec_options[k][i]
+            for k in ["wj", "radelem"]  # ["zblz", "wj", "radelem"]
+            for i, v in enumerate(bispec_options[k])
+        }
+    )
+    return d
+
+
+def set_variables(lmp, **lmp_variable_args):
+    for k, v in lmp_variable_args.items():
+        lmp.command(f"variable {k} equal {v}")
+
+
+def set_computes_snappy(lmp, bispec_options):
+    # # Bispectrum coefficient computes
+    base_b = "compute b all sna/atom ${rcutfac} ${rfac0} ${twojmax}"
+    base_db = "compute db all snad/atom ${rcutfac} ${rfac0} ${twojmax}"
+    base_vb = "compute vb all snav/atom ${rcutfac} ${rfac0} ${twojmax}"
+
+    numtypes = bispec_options["numtypes"]
+    radelem = " ".join([f"${{radelem{i}}}" for i in range(1, numtypes + 1)])
+    wj = " ".join([f"${{wj{i}}}" for i in range(1, numtypes + 1)])
+
+    kw_options = {
+        k: bispec_options[v]
+        for k, v in {
+            # "diagonal": "diagonalstyle", For legacy diagonalstyle option
+            "rmin0": "rmin0",
+            "bzeroflag": "bzeroflag",
+            "quadraticflag": "quadraticflag",
+            "switchflag": "switchflag",
+        }.items()
+        if v in bispec_options
+    }
+    kw_substrings = [f"{k} {v}" for k, v in kw_options.items()]
+    kwargs = " ".join(kw_substrings)
+
+    for op, base in zip(("b", "db", "vb"), (base_b, base_db, base_vb)):
+        # print("Setting up compute",op)
+        command = f"{base} {radelem} {wj} {kwargs}"
+        # print(command)
+        lmp.command(command)
+
+    for cname in ["b", "db", "vb"]:
+        lmp.command(f"compute {cname}_sum all reduce sum c_{cname}[*]")
+
+
+def extract_computes_snappy(lmp, num_atoms, n_coeff, num_types):
+    lmp_atom_ids = lmp.numpy.extract_atom_iarray("id", num_atoms).flatten()
+    assert np.all(
+        lmp_atom_ids == 1 + np.arange(num_atoms)
+    ), "LAMMPS seems to have lost atoms"
+
+    # Extract types
+    lmp_types = lmp.numpy.extract_atom_iarray(name="type", nelem=num_atoms).flatten()
+    lmp_volume = lmp.get_thermo("vol")
+
+    # Extract Bsum
+    lmp_bsum = extract_compute_np(lmp, "b_sum", 0, 1, (n_coeff))
+
+    # Extract B
+    lmp_barr = extract_compute_np(lmp, "b", 1, 2, (num_atoms, n_coeff))
+
+    type_onehot = np.eye(num_types)[lmp_types - 1]  # lammps types are 1-indexed.
+    # has shape n_atoms, n_types, num_coeffs.
+    # Constructing so it has the similar form to db and vb arrays. This adds some memory usage,
+    # but not nearly as much as vb or db (which are factor of 6 and n_atoms*3 larger, respectively)
+
+    b_atom = type_onehot[:, :, np.newaxis] * lmp_barr[:, np.newaxis, :]
+    b_sum = b_atom.sum(axis=0) / num_atoms
+
+    try:
+        assert np.allclose(
+            lmp_bsum, lmp_barr.sum(axis=0), rtol=1e-12, atol=1e-12
+        ), "b_sum doesn't match sum of b"
+    except AssertionError:
+        print(lmp_bsum)
+        print(lmp_barr.sum(axis=0))
```

Comment:
Seems like old debugging code, remove if not needed or raise an exception.

Reply:
I removed the check.